src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/index.tsx

```diff
@@ -1,83 +1,205 @@
 import React from 'react';
+import {
+  DragDropContext,
+  DragDropContextProps,
+  Draggable,
+  DragStart,
+  DragUpdate,
+  Droppable,
+  DropResult,
+} from 'react-beautiful-dnd';
 import styled from '@emotion/styled';
 
 import {PanelTable} from 'app/components/panels';
 import {t} from 'app/locale';
-import overflowEllipsis from 'app/styles/overflowEllipsis';
 import {DynamicSamplingRule} from 'app/types/dynamicSampling';
 
+import Placeholder from './placeholder';
 import Rule from './rule';
+import {getClientY, getDraggedDom, layout} from './utils';
+
+type DragEndChildrenProps = Required<Parameters<DragDropContextProps['onDragEnd']>[0]>;
 
 type Props = {
   rules: Array<DynamicSamplingRule>;
   onEditRule: (rule: DynamicSamplingRule) => () => void;
   onDeleteRule: (rule: DynamicSamplingRule) => () => void;
+  onDragEnd: (result: DragEndChildrenProps) => void;
   disabled: boolean;
 };
 
-function Rules({rules, onEditRule, onDeleteRule, disabled}: Props) {
-  return (
-    <StyledPanelTable
-      headers={['', t('Event Type'), t('Category'), t('Sampling Rate'), '']}
-      isEmpty={!rules.length}
-      emptyMessage={t('There are no rules to display')}
-    >
-      {rules.map((rule, index) => (
-        <Rule
-          key={index}
-          rule={rule}
-          onEditRule={onEditRule(rule)}
-          onDeleteRule={onDeleteRule(rule)}
-          disabled={disabled}
-        />
-      ))}
-    </StyledPanelTable>
-  );
-}
+type State = {
+  placeholderProps?: {
```

Comment:
I think we should reconsider if we need a placeholder. The new library does a good job of providing feedback on where an item will end up, and we could use background colors to create contrast between the gap and other items.

Reply:
yes, that is a difficult decision. I like the dashed placeholder, but I agree with you about the complexity that it brings